builder/config/load.go

```diff
@@ -0,0 +1,32 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+
+	"github.com/uor-framework/client/builder/api/v1alpha1"
+)
+
+func ReadConfig(configName string) (c v1alpha1.DataSetConfiguration, err error) {
+
+	viper.SetConfigName(configName)
+	viper.AddConfigPath(".")
+	viper.SetConfigType("yaml")
+
+	var configuration v1alpha1.DataSetConfiguration
+
+	err = viper.ReadInConfig()
```

Comment:
This might need a little reword because you are trying to Unmarshal before fully processing the error for `ReadInConfig`.

Reply:
the errors are now returned when able